src/openforms/authentication/utils.py

```diff
@@ -106,6 +106,30 @@ def get_cosign_login_url(request: Request, form: Form, plugin_id: str) -> str:
     return auth_page.url
 
 
+def same_cosigner(request: Request, submission: Submission) -> bool:
+    from openforms.submissions.utils import is_same_login_attribute, is_same_login_value
+
+    if not submission.is_authenticated:
+        logger.info(
+            "cosign_not_allowed",
+            reason="missing_auth_info",
+            submission_uuid=submission.uuid,
+        )
+        return False
+
+    if is_same_login_value(submission, request) and is_same_login_attribute(
+        submission, request
+    ):
+        logger.info(
+            "cosign_not_allowed",
+            reason="cannot_sign_own_cosign_request",
+            submission_uuid=submission.uuid,
+        )
+        return True
```

Comment:
This currently only verifies that the original form submitter is not the same person trying to cosign the submission by checking if the login method used (DigiD/eHerkenning) and the saved value does not match with what is used in the session during cosigning.

This does not (at the moment):
- Verify that whenever another login method is used (for example eHerkenning), compared to the original submission (for example DigiD), that both values match
- Verify that the user trying to cosign, has the same emailadres as was given during the form submission (see second last point from https://github.com/open-formulieren/open-forms/issues/4459#issuecomment-2612168297)